Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs

```diff
@@ -35,39 +37,36 @@ public void Execute(in ArchetypeChunk chunk, int unfilteredChunkIndex, bool useE
 
             velNew.y += gravityDt;
             pos += TimeStep * velNew;
-            float notOnGround = Convert.ToSingle(pos.y > 0);
+            bool notOnGround = pos.y < FieldArea.Value.Min.y;
 
-            if (notOnGround != 1.0f && velNew.y < -k_DeepImpactVel) // deep impact
+            if (notOnGround && velNew.y < -k_DeepImpactVel) // deep impact
             {
                 velNew.y = -velNew.y * k_Restitution;
-                velNew.x *= k_FrictionDamping;
-                velNew.z *= k_FrictionDamping;
+                velNew.xz *= k_FrictionDamping;
             }
             else
             {
-                velNew.y *= notOnGround;
-                velNew.x *= notOnGround;
-                velNew.z *= notOnGround;
+                velNew = math.select(velNew,float3.zero, notOnGround);
             }
 
-            // stop at lower bottom
-            pos.y *= notOnGround;
+            // clamp to field
+            pos = math.clamp(pos, FieldArea.Value.Min, FieldArea.Value.Max);
 
             tmComp.Value.Position = pos;
             velComp.Value = velNew;
 
             // derive heading of velocity
             var heading = math.atan2(velNew.x, velNew.z);
-            var headingRot = Quaternion.AngleAxis(math.degrees(heading), Vector3.up);
+            var headingRot = quaternion.AxisAngle(math.up(),math.degrees(heading));
 
             // derive pitch of velocity, by bringing velocity into local space of object (with z forward) and
             // calculating the angle of the velocity in the local yz plane of the object
             var velForwardLocal = math.rotate(math.inverse(headingRot), velNew);
             var pitch = math.atan2(velForwardLocal.y, velForwardLocal.z);
-            var pitchRot = Quaternion.AngleAxis(math.degrees(pitch), Vector3.left);
+            var pitchRot = quaternion.AxisAngle(math.left(),math.degrees(pitch));
 
             // final rotation by applying pitch first and then heading (right to left) 
-            tmComp.Value.Rotation = headingRot * pitchRot;
+            tmComp.Value.Rotation = math.mul(headingRot, pitchRot);
```

Comment:
Would it make a difference in Burst if you don't use math.mul here?
I hoped it wouldn't... Would love to know for sure though.